nlp_apps.ipynb

```diff
@@ -797,11 +772,251 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As you might have observed that our dataset is equally balanced i.e. we have an equal number of words in both the classes."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Step 4:-Making the Naive Bayes Model**\n",
+    "\n",
+    "Using Naive Bayes classifier we can calculate the probability of a word in `company` and `fruit` class and then multiplying all of them to get the probability of that sentence belonging each of the given classes. But if a word is not in our dictionary then this leads to the probability of that class becoming zero. For eg:- the word *Foxconn* is not in the dictionary of any of the classes. Due to this the \n",
+    "\n",
+    "To solve the problem we need to use **smoothing**, i.e. providing a minimum non-zero threshold probability to every word that we come across.\n",
+    "\n",
+    "The `UnigramWordModel` class has implemented smoothing by taking an additional argument from the user i.e. the minimum frequency that we would be giving to every word even if it is new to the dictionary."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_words_0 = UnigramWordModel(words_0, 1)\n",
+    "model_words_1 = UnigramWordModel(words_1, 1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we would be making the Naive Bayes model. For that, we would be making `dist` as we had done earlier in the Authorship Recognition Task."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from learning import NaiveBayesLearner\n",
+    "\n",
+    "dist = {('company', 1): model_words_0, ('fruit', 1): model_words_1}\n",
+    "\n",
+    "nBS = NaiveBayesLearner(dist, simple = True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Step 5:- Predict the class of the label**\n",
+    "\n",
+    "Now we would be making a function that does pre-procesing of the sentences which we have taken for testing. And then predicting the class of every sentence in the document."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def recognize(sentence, nBS):\n",
+    "    sentence_words = words(sentence)\n",
+    "    return nBS(sentence_words)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Apple Inc. supplier Foxconn demos its own iPhone-compatible smartwatch\t-company\n",
+      "I now know how to make a delicious apple pie thanks to the best teachers ever\t-fruit\n"
+     ]
+    }
+   ],
+   "source": [
+    "# prediction the class of every sentence in the test set\n",
```

Comment:
What do you mean by 'prediction the class of every sentence in the test set'? The sentence is no grammatically correct and it is a bit confusing.